src/kernel/filesystem/fat32.zig

```diff
@@ -1927,18 +1988,52 @@ test "Fat32FS.write" {
 test "Fat32FS.open - no create" {
     var stream = &std.io.fixedBufferStream(test_stream_buff[0..]);
 
-    try mkfat32.Fat32.make(.{}, stream);
+    try mkfat32.Fat32.make(.{}, stream, true);
 
     var test_fs = try initialiseFAT32(std.testing.allocator, stream);
     defer test_fs.deinit();
 
-    // TODO: Once the open PR is done
+    // Temporary add hand built files to open
+    // Once write PR is done, then can test real files
+    var entry_buff = [_]u8{
+        // Long entry 3
+        0x43, 0x6E, 0x00, 0x67, 0x00, 0x6E, 0x00, 0x61, 0x00, 0x6D, 0x00, 0x0F, 0x00, 0x6E, 0x65, 0x00,
+        0x2E, 0x00, 0x74, 0x00, 0x78, 0x00, 0x74, 0x00, 0x00, 0x00, 0x00, 0x00, 0xFF, 0xFF, 0xFF, 0xFF,
+        // Long entry 2
+        0x02, 0x6E, 0x00, 0x67, 0x00, 0x76, 0x00, 0x65, 0x00, 0x72, 0x00, 0x0F, 0x00, 0x6E, 0x79, 0x00,
+        0x6C, 0x00, 0x6F, 0x00, 0x6F, 0x00, 0x6F, 0x00, 0x6F, 0x00, 0x00, 0x00, 0x6F, 0x00, 0x6F, 0x00,
+        // Long entry 1
+        0x01, 0x6C, 0x00, 0x6F, 0x00, 0x6F, 0x00, 0x6F, 0x00, 0x6F, 0x00, 0x0F, 0x00, 0x6E, 0x6F, 0x00,
+        0x6E, 0x00, 0x67, 0x00, 0x6C, 0x00, 0x6F, 0x00, 0x6F, 0x00, 0x00, 0x00, 0x6F, 0x00, 0x6F, 0x00,
+        // Short entry
+        0x4C, 0x4F, 0x4F, 0x4F, 0x4F, 0x4F, 0x7E, 0x31, 0x54, 0x58, 0x54, 0x00, 0x18, 0xA0, 0x68, 0xA9,
+        0xFE, 0x50, 0x00, 0x00, 0x00, 0x00, 0x6E, 0xA9, 0xFE, 0x50, 0x08, 0x00, 0x13, 0x00, 0x00, 0x00,
+        // Long entry 2
+        0x42, 0x2E, 0x00, 0x74, 0x00, 0x78, 0x00, 0x74, 0x00, 0x00, 0x00, 0x0F, 0x00, 0xE9, 0xFF, 0xFF,
+        0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0x00, 0x00, 0xFF, 0xFF, 0xFF, 0xFF,
+        // Long entry 1
+        0x01, 0x72, 0x00, 0x61, 0x00, 0x6D, 0x00, 0x64, 0x00, 0x69, 0x00, 0x0F, 0x00, 0xE9, 0x73, 0x00,
+        0x6B, 0x00, 0x5F, 0x00, 0x74, 0x00, 0x65, 0x00, 0x73, 0x00, 0x00, 0x00, 0x74, 0x00, 0x31, 0x00,
+        // Short entry
+        0x52, 0x41, 0x4D, 0x44, 0x49, 0x53, 0x7E, 0x31, 0x54, 0x58, 0x54, 0x00, 0x18, 0x34, 0x47, 0x76,
+        0xF9, 0x50, 0x00, 0x00, 0x00, 0x00, 0x48, 0x76, 0xF9, 0x50, 0x03, 0x00, 0x10, 0x00, 0x00, 0x00,
+    };
+
+    // Goto root dir and write a long and short entry
+    const sector = test_fs.fat_config.clusterToSector(test_fs.root_node.cluster);
+    try test_fs.stream.seekableStream().seekTo(sector * test_fs.fat_config.bytes_per_sector);
+    try test_fs.stream.writer().writeAll(entry_buff[0..]);
+
+    vfs.setRoot(test_fs.root_node.node);
+
+    const file = try vfs.openFile("/ramdisk_test1.txt", .NO_CREATION);
```

Comment:
Does `file` need to be checked at all? Such as it pointing to the correct thing entry (like how the test fs checks that it is the same as what is in the file tree)?

Reply:
can add extra tests